testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -299,6 +312,60 @@ public void testViewColumnAccessControl()
         assertAccessAllowed(viewOwnerSession, "DROP VIEW " + invokerViewName);
     }
 
+    @Test
+    public void testViewOwnersRoleGrants()
+    {
+        TestingSystemSecurityMetadata systemSecurityMetadata = (TestingSystemSecurityMetadata) getDistributedQueryRunner().getCoordinator().getInstance(Key.get(SystemSecurityMetadata.class));
+        Session viewOwnerSession = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.forUser("test_view_access_owner")
+                        .withEnabledRoles(Set.of("ownerrole"))
+                        .build())
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+
+        String columnAccessViewName = "test_view_column_access_" + randomNameSuffix();
+
+        systemSecurityMetadata.grantRoles(getSession(), Set.of("ownerrole"), Set.of(new TrinoPrincipal(USER, "test_view_access_owner")), false, Optional.empty());
+        systemSecurityMetadata.setViewOwner(
+                getSession(),
+                new CatalogSchemaTableName("blackhole", "default", columnAccessViewName),
+                new TrinoPrincipal(USER, viewOwnerSession.getUser()));
+
+        assertAccessAllowed(
+                viewOwnerSession,
+                "CREATE VIEW " + columnAccessViewName + " AS SELECT * FROM orders",
+                privilege("orders", CREATE_VIEW_WITH_SELECT_COLUMNS));
+
+        // verify that role permissions are taken for the view owner's role grants
+        // even if executed by the owner with different roles enabled
+        executeExclusively(() -> {
```

Comment:
I think this class fakes parallel execution. It would run faster if it was just singlethreaded.